libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -195,65 +189,51 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
     }
 
     private fun sessionStart() {
-        var freeDrive = false
+        var isActiveGuidance = true
         ifNonNull(weakMapboxNavigation.get()?.getRoutes()) { routes ->
-            freeDrive = routes.isEmpty()
+            isActiveGuidance = routes.isNotEmpty()
         }
-        when (freeDrive) {
+        when (isActiveGuidance) {
             true -> {
-                // Do nothing
-                Log.d(TAG, "Free drive not supported")
-            }
-            false -> {
                 telemetryThreadControl.scope.launch {
                     callbackDispatcher.resetRouteProgressProcessor()
                     postUserEventDelegate =
                         postUserFeedbackEventAfterInit // Telemetry is initialized and the user selected a route. Allow user feedback events to be posted
                     handleSessionStart()
-                    sessionEndPredicate = { sessionStop() }
+                    sessionEndPredicate = {
+                        telemetryScope.launch {
+                            sessionStop()
+                        }
+                    }
                 }
             }
+            false -> {
+                // Do nothing
+                Log.d(TAG, "Only Active Guidance supported")
+            }
         }
     }
 
-    private fun sessionStop() {
-        Log.d(TAG, "TripSessionState.STOPPED")
+    private suspend fun sessionStop() {
+        Log.d(TAG, "sessionStop")
         postUserEventDelegate =
             postUserEventBeforeInit // The navigation session is over, disallow posting user feedback events
         when (dynamicValues.routeArrived.get()) { // Cancellation events will be sent after an arrival event.
             true, false -> {
-                telemetryThreadControl.scope.launch {
+                telemetryScope.launch {
```

Comment:
is this needed? looks like this function is already in a `suspend`

Reply:
hmm maybe this is why the supervisor job is needed, to handle the multiple children?
https://medium.com/androiddevelopers/exceptions-in-coroutines-ce8da1ec060c

launch -> launch calls are hard to wrap the head around

Reply:
It's needed for `join()` 👀 comment above https://github.com/mapbox/mapbox-navigation-android/pull/3162/files#r439164901